src/components/viewmodels/roomlist/RoomListItemViewModel.tsx

```diff
@@ -190,3 +201,35 @@ function getA11yLabel(roomName: string, notificationState: RoomNotificationState
         return _t("room_list|room|open_room", { roomName });
     }
 }
+
+function useRoomMessagePreview(room: Room): string | undefined {
+    const { shouldShowMessagePreview } = useMessagePreviewToggle();
+    const [previewText, setPreviewText] = useState<string | undefined>(undefined);
+
+    const updatePreview = useCallback(async () => {
+        if (!shouldShowMessagePreview) {
+            setPreviewText(undefined);
+            return;
+        }
+
+        const roomIsDM = Boolean(DMRoomMap.shared().getUserIdForRoomId(room.roomId));
+        // For the tag, we only care about whether the room is a DM or not as we don't show
+        // display names in previewsd for DMs, so anything else we just say is 'untagged'
+        // (even though it could actually be have other tags: we don't care about them).
+        const messagePreview = await MessagePreviewStore.instance.getPreviewForRoom(
+            room,
+            roomIsDM ? DefaultTagID.DM : DefaultTagID.Untagged,
+        );
+        if (messagePreview) setPreviewText(messagePreview.text);
+    }, [room, shouldShowMessagePreview]);
+
+    useEventEmitter(MessagePreviewStore.instance, MessagePreviewStore.getPreviewChangedEventName(room), () => {
```

Comment:
Please use `useTypedEventEmitter`

Reply:
This would need the AsyncStore stuff converting to TypedEventEmitter I think, which seems like too much to squeeze into this PR?